packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -55,32 +44,40 @@ const LinkControlSearchInput = ( {
 
 	return (
 		<form onSubmit={ selectSuggestionOrCurrentInputValue }>
-			<URLInput
-				className="block-editor-link-control__search-input"
-				value={ value }
-				onChange={ selectItemHandler }
-				onFocus={ selectItemHandler }
-				onKeyDown={ ( event ) => {
-					if ( event.keyCode === ENTER ) {
-						return;
+			<div className="block-editor-link-control__search-input-wrapper">
+				<URLInput
+					className="block-editor-link-control__search-input"
+					value={ value }
+					onChange={ selectItemHandler }
+					onFocus={ selectItemHandler }
+					placeholder={ __( 'Search or type url' ) }
+					__experimentalRenderSuggestions={ renderSuggestions }
+					__experimentalFetchLinkSuggestions={ fetchSuggestions }
+					__experimentalHandleURLSuggestions={ true }
+					__experimentalShowInitialSuggestions={
+						showInitialSuggestions
 					}
-					handleLinkControlOnKeyDown( event );
-				} }
-				onKeyPress={ handleLinkControlOnKeyPress }
-				placeholder={ __( 'Search or type url' ) }
-				__experimentalRenderSuggestions={ renderSuggestions }
-				__experimentalFetchLinkSuggestions={ fetchSuggestions }
-				__experimentalHandleURLSuggestions={ true }
-				__experimentalShowInitialSuggestions={ showInitialSuggestions }
-			/>
-			<div className="block-editor-link-control__search-actions">
-				<Button
-					type="submit"
-					label={ __( 'Submit' ) }
-					icon="editor-break"
-					className="block-editor-link-control__search-submit"
 				/>
+				<div className="block-editor-link-control__search-actions">
+					<Button
+						type="submit"
+						label={ __( 'Submit' ) }
+						icon="editor-break"
+						className="block-editor-link-control__search-submit"
+					/>
+				</div>
 			</div>
+
+			{ errorMessage && (
+				<Notice
+					className="block-editor-link-control__search-error"
+					status="error"
+					isDismissible={ false }
+					politeness="assertive"
```

Comment:
It's not strictly necessary to assign `politeness` here, because `Notice` tries to be "smart" with its defaults, and will use `assertive` when `status` is `error`. 

https://github.com/WordPress/gutenberg/blob/master/packages/components/src/notice/README.md#props

>If not provided, a sensible default is used based on the notice status.

It's not explicitly prescribed what this default would be, but I also don't think it needs to be or should be, considering the purpose of the default is to make this decision on your behalf to align best to the accessibility recommendations.

Reply:
Ok I've removed this and gone with your recommendation.